### ISSUE TITLE
Redesign proposal for the navbar

### ISSUE THREAD

**Issue by julesverhaeren** [fguillot/kanboard#1807]
I hope I am not insulting someone when I say that I think that the current navbar looks a bit messy and that it could use an update. I've made quick responsive  [demo](http://julesverhaeren.github.io/kanboardredesignproposal/) ( http://julesverhaeren.github.io/kanboardredesignproposal/ ) of what I was thinking of. This design still has some quirks and has only been made for the board view, as it is only meant for showing how I'd imagine a new navbar. What are your thoughts?

**Comment by fguillot** [comment]
It's very nice! I may steal some ideas :)

However, we need to have some room for the project title that can be longer and the board switcher on the right.

The top header and the project page header are not the same templates, that will require more complex changes.

But I like the search box with the dropdowns, this one can be implemented easily for the next version.

**Comment by julesverhaeren** [comment]
Thanks!

I was trying to fit everything in a single row but that probably was too optimistic. Perhaps I should aim for smaller more easily implementable improvements. Have there been made any design guidelines? I couldn't find them in the documentation. 

For example with the dropdowns, is there a reason the "Actions" , "Filters" and "Display another project" buttons all have a different styling (border/no border, gradient/no gradient)? I think it would look much cleaner if the looks of elements were unified.

**Comment by BlueTeck** [comment]
:+1:

**Comment by Busfreak** [comment]
:thumbsup:

**Comment by bdrayer** [comment]
:+1:

**Comment by fguillot** [comment]
For your information the search box with filters have been implemented in the development version.

**Comment by mcorteel** [comment]
@fguillot I am interested in improving the design of kanboard and the first major thing that gets in the way (for the toolbar) is the fact that icons are not contained in the `<a>` tag.

The idea I had for the `URL::link()` method was to detect a string like `<fa-user>` in the `$label` string and replace it with `<i class="fa fa-user fa-fw"></i>`.

It wouldn't impact code readabilty IMHO, it would be completely backward-compatible and it would make a lot more sense if icons were contained in the anchor. This would also allow a much better styling.

**Comment by mcorteel** [comment]
I did a proof-of-concept for this:
![spectacle h21833](https://cloud.githubusercontent.com/assets/6164603/13566018/b86adc5e-e453-11e5-9f86-444feab2d4aa.png)

I think it's leaner (but that's just me, what do you think?). This style could be used to make headers more homogenous. It also saves about 10px of valuable vertical space.

If you are interested, I can send a PR.

**Comment by julesverhaeren** [comment]
@mcorteel Wow that looks really nice! I think it does become a bit crowded however, perhaps you could use (some of) those saved vertical pixels to create some space between the buttons and the column titles?

**Comment by BlueTeck** [comment]
@mcorteel :+1:

**Comment by mcorteel** [comment]
Thank! Yeah, I increased the margin a little. I also increased the font size because it was a little harder to read.

![spectacle l17159](https://cloud.githubusercontent.com/assets/6164603/13567641/b30c28e0-e45c-11e5-99d3-091799279839.png)

I think it really helps with homogeneity too:

![spectacle e23226](https://cloud.githubusercontent.com/assets/6164603/13568209/03ad5b18-e460-11e5-9588-c96a42506234.png)
